lib/cpl.rb

```diff
@@ -181,7 +182,9 @@ def self.all_base_commands
 
           show_info_header(config) if with_info_header
 
-          command_class.new(config).call
+          the_command = command_class.new(config)
+          the_command.prepare_extended_config unless with_minimal_config
+          the_command.call
```

Comment:
I don't really like this implementation.

Why don't we lazy-load the config like this: we change `apps`, `app`, `org` and `current` to be methods, and then if a method is called for the first time, we load the data needed.

For example, for app:

```rb
def app
  return @app if @app_loaded

  ...
  @app_loaded = true
end
```

Then we don't need `WITH_MINIMAL_CONFIG`, as it will just work everywhere.

Reply:
And how we control calling `load_app_config` and `load_apps` in `initialize` method of `Config` class? Those are the issue.

Reply:
Done